packages/file_selector/file_selector_windows/lib/src/file_selector_dart/shell_win32_api.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+
+/// A thin wrapper for Win32 platform specific Shell methods.
+///
+/// The only purpose of this class is to decouple specific Win32 Api call from the bussiness logic so it can be init tested in any environment.
+class ShellWin32Api {
+  /// Creates and [initializes](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-shcreateitemfromparsingname) a Shell item object from a parsing name.
+  /// If the directory doesn't exist it will return an error result.
+  int createItemFromParsingName(
+    String initialDirectory,
+    Pointer<GUID> ptrGuid,
+    Pointer<Pointer<NativeType>> ptrPath,
+  ) {
+    return using((Arena arena) {
+      return SHCreateItemFromParsingName(
+          initialDirectory.toNativeUtf16(allocator: arena),
```

Comment:
`ptrGuid` and `ptrPath` are allocated before this method is called, but the native UTF-16 string is allocated in this method. Would it be better to simplify this by also allocating the native UTF-16 string before calling this method?

Reply:
Yes, like that, applied.